zeebe/atomix/utils/src/main/java/io/atomix/utils/Version.java

```diff
@@ -16,29 +16,33 @@
  */
 package io.atomix.utils;
 
-import static com.google.common.base.Preconditions.checkArgument;
-import static java.lang.Integer.parseInt;
-
-import com.google.common.collect.ComparisonChain;
-import java.util.Arrays;
+import io.camunda.zeebe.util.SemanticVersion;
 import java.util.Objects;
 
-/** Atomix software version. */
+/**
+ * Atomix software version.
+ *
+ * <p>NOTE: eventually we should stop using this and use {@link SemanticVersion} directly instead.
+ */
 public final class Version implements Comparable<Version> {
 
   private final int major;
   private final int minor;
   private final int patch;
   private final String build;
-
-  private Version(final int major, final int minor, final int patch, final String build) {
-    checkArgument(major >= 0, "major version must be >= 0");
-    checkArgument(minor >= 0, "minor version must be >= 0");
-    checkArgument(patch >= 0, "patch version must be >= 0");
+  private final String metadata;
```

Comment:
:thought_balloon: Backwards compatibility is preserved, since we only add a field, which Kryo can deal with - and that field can be null, it's fine.

Reply:
It is only an issue if an old broker receives a version from an updated broker right? So via topology? 🤔